devtools/create_xeen/constants.cpp

```diff
@@ -1708,625 +1881,656 @@ static const int RACE_COLD_RESISTENCES[5] = { 7, 0, 2, 5, 10 };
-
-static const char *const GIVE_TREASURE_FORMATTING =
-	"\x3l\v060\t000\x4""077\n"
-	"\x4""077\n"
-	"\x4""077\n"
-	"\x4""077\n"
-	"\x4""077\n"
-	"\x4""077";
-
-static const char *const X_FOUND_Y = "\v060\t000\x3""c%s found: %s";
-
-static const char *const ON_WHO = "\x3""c\v009On Who?";
-
-static const char *const WHICH_ELEMENT1 =
-	"\r\x3""c\x1Which Element?\x2\v034\t014\f15F\fdire\t044"
-	"\f15E\fdlec\t074\f15C\fdold\t104\f15A\fdcid\x1";
-
-static const char *const WHICH_ELEMENT2 =
-	"\r\x3""cWhich Element?\x2\v034\t014\f15F\fdire\t044"
-	"\f15E\fdlec\t074\f15C\fdold\t104\f15A\fdcid\x1";
-
-static const char *const DETECT_MONSTERS = "\x3""cDetect Monsters";
-
-static const char *const LLOYDS_BEACON =
-	"\r\x3""c\v000\t000\x1Lloyd's Beacon\n"
-	"\n"
-	"Last Location\n"
-	"\n"
-	"%s\x3l\n"
-	"x = %d\x3r\t000y = %d\x3""c\x2\v122\t021\f15S\fdet\t060\f15R\fdeturn\x1";
-
-static const char *const HOW_MANY_SQUARES = "\x3""cTeleport\nHow many squares %s (1-9)\n";
-
-static const char *const TOWN_PORTAL =
-	"\x3""cTown Portal\x3l\n"
-	"\n"
-	"\t0101. %s\n"
-	"\t0102. %s\n"
-	"\t0103. %s\n"
-	"\t0104. %s\n"
-	"\t0105. %s\x3""c\n"
-	"\n"
-	"To which Town (1-5)\n"
-	"\n";
-
-static const char *const TOWN_PORTAL_SWORDS =
-	"\x3""cTown Portal\x3l\n"
-	"\n"
-	"\t0101. %s\n"
-	"\t0102. %s\n"
-	"\t0103. %s\x3""c\n"
-	"\n"
-	"To which Town (1-3)\n"
-	"\n";
+  { -1, -1, -1, -1 },
+  { 21, -1, -1, -1 },
+  { 22, -1, -1, -1 },
+  { 21, 1, 14, -1 },
+  { 22, 0, 25, -1 },
+  { -1, -1, -1, -1 },
+  { -1, -1, -1, -1 },
+  { -1, -1, -1, -1 },
+  { 20, 1, 11, 23 },
+  { 20, 1, -1, -1 }
+};
+
+static const char* const COMBAT_DETAILS = "\r\f00\x3""c\v000\t000\x2%s%s%s\x1";
+
+static const char* NOT_ENOUGH_TO_CAST =
+  "\x3""c\v000�������筮 %s, �⮡� ��������� %s";
+
+static const char* SPELL_CAST_COMPONENTS[2] = { "�窮� �����", "�������" };
+
+static const char* const CAST_SPELL_DETAILS =
+  "\r\x2\x3""c\v122\t013\f37�\fd���\t040\f37�\fd��"
+  "\t067ESC\x1\t000\v000\x3""c����������\n"
+  "\n"
+  "%s\x3""c\n"
+  "\n"
+  "     ��⮢��:\x3""c\n"
+  "\n"
+  "\f09%s\fd\x2\x3""l\n"
+  "\v082����\x3""r\t000%u/%u\x3""l\n"
+  "��� ��\x3""r\t000%u\x1";
+
+static const char* const PARTY_FOUND =
+  "\x3""c�������:\n"
+  "\n"
+  "\x3""r\t000%lu ���\n"
+  "%lu ����";
+
+static const char* const BACKPACKS_FULL_PRESS_KEY =
+  "\v007\f12��������! ���� ��९������!\fd\n"
+  "������ �������";
+
+static const char* const HIT_A_KEY =
+  "\x3""l\v120\t000\x4""077\x3""c\f37�. �������\fd";
+
+static const char* const GIVE_TREASURE_FORMATTING =
+  "\x3l\v060\t000\x4""077\n"
+  "\x4""077\n"
+  "\x4""077\n"
+  "\x4""077\n"
+  "\x4""077\n"
+  "\x4""077";
+
+static const char* const X_FOUND_Y = "\v060\t000\x3""c%s ���%s: %s";
+
+static const char* const ON_WHO = "\x3""c\v009����?";
+
+static const char* const WHICH_ELEMENT1 =
+  "\r\x3""c\x1����� ����?\x2\v034\t014\f15�\fd����\t044"
+  "\f15�\fd஬\t074\f15�\fd����\t104\f15�\fd�\x1";
```

Comment:
There seems to be more changes to constants.cpp than there should be just for adding Russian support. And, more worryingly, the various text strings seem to have been corrupted with non-ascii sequences like "d®«®¤".

Reply:
Fixed in later commits